### PR TITLE
WIP: Add SLP check #371

### PR DESCRIPTION
**Pull request by scarabeusiv**

### CODE REVIEW

**Comment by Conan-Kudo** [comment]
@scarabeusiv Don't we already have a variation of this?

File: rpmlint/checks/SharedLibraryPolicyCheck.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        # Verify non-lib stuff does not add dependencies
+        if libs:
+            for dep in pkg_requires.difference(_essential_dependencies):
+                if '.so.' in dep and dep not in libs and dep not in libs_needed:
+                    printError(pkg, 'shlib-policy-excessive-dependency', dep)
+
+        # Check for non-versioned directories beyond sysdirs in package
+        sysdirs = ('/lib', '/lib64', '/usr/lib', '/usr/lib64',
+                   '/usr/share', '/usr/share/licenses',
+                   '/usr/share/doc/packages')
```
Comment (Conan-Kudo) [inline rpmlint/checks/SharedLibraryPolicyCheck.py:300]: This should be `/usr/share/doc` instead of `/usr/share/doc/packages`

**Comment by scarabeusiv** [comment]
Nope the SLP is not tested elsewhere. There is the weird %mklibname in speccheck.

In BinaryCheck we have test for the soname and if the library filename matches that only.

File: rpmlint/checks/SharedLibraryPolicyCheck.py (hunk 1 above)
Comment (scarabeusiv) [reply rpmlint/checks/SharedLibraryPolicyCheck.py:300]: Yep, there is also quite bit of /opt and kde3 prefixes that must go away :)

**Comment by Conan-Kudo** [comment]
`%mklibname` check is the Mandriva SLP enforcement package macro: https://wiki.mageia.org/en/Libraries_policy

**Comment by scarabeusiv** [comment]
Finished up in #560

**Pull request closed without merging**